rfcs/text/Fusillade in the DCP.md

```diff
@@ -0,0 +1,105 @@
+1. As a component of the DCP, I would like to define a group of users to manage the permission of a large groups of 
+users easily.
+1. As a component of the DCP, I would like to give permission to unauthenticated users.
+1. As a component of the DCP, I would like to use service account to programmatically manage my service.  
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+### As a component of the DCP, how do I control access to my resources?
+1. Define what actions and resources you'd like to control. Actions should take the form `{component}:{Action}`, for example to get a bundle from the DSS the action could be `dss:GetBundle`. The action is case sensitive, and camel case is suggested. Resources are what actions affect. The same action can be applied to multiple different resources. For the resource use the AWS ARN format. For example...
+
+1. Define the different types of users you need to support based on what actions and resource they will need permission for. These users types you define will be your roles. 
+
+1. Create a file **{role}.json** for each user type. Using the resource and actions previously define, write a policy in the AWS IAM policy format that express what each role can do. If it isn't explicitly stated, it is assumed the user with this role does not have permission. For the resource field, `*` can be used as a wild card in the same way it's used in AWS IAM policies.
+
+1. Run the script `??? -path "path to roles" -component "name of the component" -stage "current deployment stage" -secret "secretId of a service account in AWS" -service "path to service account credentials"` to add your roles to fusillade. The role that is added to fusillade will be named `{component}-{stage}-{role}`. The stage is not included for *prod* deployments.
+
+1. Users can now be assigned to this role. However the preferred method of adding users to a role is to assign this role to a group. In the DCP, groups are manage in dcpinfra repo. 
+
+### As a DCP operator how do I had groups to the DCP?
+A service account with permission to create and modify fusillade groups is required to perform this process.
+
+1. In *humancellatlas/dcpinfra/fusillade/groups* create a file name `{group}.json` where `{group}` is the name of your group. Here is an example of the contents of a *{group}.json* file 
+```JSON
+{
+      "name": "name_of_the_group",
+      "description": "a description of the group",
+      "members": [
+        "a_list_of_user_emails@example.example",
+        "a_list_of_service_account_emails@example.example"
+      ],
+      "roles": [
+        "the_roles_assigned_to_this_group"
+      ]
+}
+``` 
+
+1. In DCP infra run the script `??? -path "path to groups" -stage "current deployment stage" -secret "secretId of a 
```

Comment:
Same as the above, I'm assuming there needs to be an argument for the unique group name and a check.

Reply:
fixed